File: src/main/java/io/enthusia/express/util/Text.java
package io.enthusia.express.util;

import java.util.Map;
import org.bukkit.ChatColor;
import org.bukkit.configuration.file.FileConfiguration;

public final class Text {
  private Text() {}

  public static String color(String input) {
    return ChatColor.translateAlternateColorCodes('&', input == null ? "" : input);
  }

  public static String msg(FileConfiguration config, String key, Map<String, String> vars) {
    String prefix = config.getString("messages.prefix", "");
    String value = config.getString("messages." + key, key);
    for (Map.Entry<String, String> entry : vars.entrySet()) {
      value = value.replace("{" + entry.getKey() + "}", entry.getValue());
    }
    return color(prefix + value);
  }

  public static String msg(FileConfiguration config, String key) {
    return msg(config, key, Map.of());
  }
}

File: src/main/kotlin/io/enthusia/express/infrastructure/util/Text.kt
package io.enthusia.express.infrastructure.util

import org.bukkit.ChatColor
import org.bukkit.configuration.file.FileConfiguration

object Text {
    /** Translate configured ampersand color codes into Bukkit legacy formatting. */
    @JvmStatic
    fun color(input: String?): String = ChatColor.translateAlternateColorCodes('&', input ?: "")

    /** Apply named placeholders and the configured prefix before translating message color codes. */
    @JvmStatic
    fun msg(config: FileConfiguration, key: String, vars: Map<String, String>): String {
        val prefix = config.getString("messages.prefix", "") ?: ""
        var value = config.getString("messages.$key", key) ?: key
        for ((name, replacement) in vars) value = value.replace("{$name}", replacement)
        return color(prefix + value)
    }

    /** Apply named placeholders and the configured prefix before translating message color codes. */
    @JvmStatic
    fun msg(config: FileConfiguration, key: String): String = msg(config, key, emptyMap())
}
